sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.students.controller.saveStudentUploadedFiles.e2e-spec.ts

```diff
@@ -0,0 +1,160 @@
+
+    notificationActionsService = await module.get(NotificationActionsService);
+    appModule = module;
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("Should save all uploaded files when a proper student file upload request is made.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplication(db.dataSource, { student });
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const studentFile1 = await saveFakeStudentFileUpload(
+      db.dataSource,
+      {
+        student,
+        creator: student.user,
+      },
+      {
+        fileOrigin: FileOriginType.Student,
+        groupName: "application",
+      },
+    );
+    const studentFile2 = await saveFakeStudentFileUpload(
+      db.dataSource,
+      {
+        student,
+        creator: student.user,
+      },
+      {
+        fileOrigin: FileOriginType.Student,
+        groupName: "application",
+      },
+    );
+
+    const payload = {
+      submittedForm: {
+        documentPurpose: "application",
+        applicationNumber: application.applicationNumber,
+      },
+      associatedFiles: [
+        studentFile1.uniqueFileName,
+        studentFile2.uniqueFileName,
+      ],
+    };
+
+    const notificationMessage = await db.notificationMessage.findOne({
+      where: { id: NotificationMessageType.StudentFileUpload },
+    });
+
+    jest
+      .spyOn(
+        notificationActionsService as any,
+        "assertNotificationMessageDetails",
+      )
+      .mockReturnValue({
+        templateId: notificationMessage.templateId,
+        emailContacts: ["test@test.com"],
+      });
+
+    const endpoint = "/students/student/save-uploaded-files";
+
+    // Mock user service to return the saved student.
+    await mockUserLoginInfo(appModule, student);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(payload)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+
+    const updatedStudentFiles = await db.studentFile.find({
+      select: {
+        id: true,
+        modifier: { id: true },
+        metadata: { applicationNumber: true },
+      },
+      where: { id: studentFile1.id },
+      relations: { modifier: true },
+    });
+
+    updatedStudentFiles.forEach((studentFile) => {
+      expect(studentFile).toMatchObject({
+        id: studentFile.id,
+        modifier: { id: student.user.id },
+        metadata: {
+          applicationNumber: application.applicationNumber,
+        },
+      });
+    });
+
+    const notification = await db.notification.findOne({
+      where: {
+        messagePayload: Raw((alias) => `${alias} @> :jsonValue`, {
+          jsonValue: JSON.stringify({
+            personalisation: {
+              applicationNumber: application.applicationNumber,
+            },
+          }),
+        }),
+      },
+    });
```

Comment:
The findone can be simplified and be more particular to the student by this way. 

```ts
const notification = await db.notification.findOne({
      select: {
        id: true,
        messagePayload: true,
      },
      where: {
        user: { id: student.user.id },
      },
    });
```